sql/log_event.cc

```diff
@@ -3317,6 +3336,28 @@ int Query_log_event::do_apply_event(Relay_log_info const *rli,
     you.
   */
   thd->catalog= catalog_len ? (char *) catalog : (char *)"";
+
+  {
```

Comment:
Does original commit introduce a new block here `{...}` ?

Reply:
I added `{...}` here because there were warnings about variables declaration crossing with `goto end`. The upstream used the same solution at https://github.com/mysql/mysql-server/blob/5.5/sql/log_event.cc#L3497